src/main/java/net/greatstart/controllers/InvestmentInterestController.java

```diff
@@ -35,59 +39,60 @@ public InvestmentInterestController(InvestmentInterestService investmentInterest
-        List<InvestmentInterest> investmentInterestList = this.investmentInterestService
-                .getAllInvestmentInterest();
-        ModelAndView model = new ModelAndView("invinterest/invinterests");
-        model.addObject("inv_interest_list", investmentInterestList);
-        return model;
+    @GetMapping()
+    public ResponseEntity<List<DtoInterest>> getAllInterests() {
+        List<DtoInterest> interests = investmentInterestService.getAllDtoInterest();
+        return new ResponseEntity<>(interests, HttpStatus.OK);
     }
 
-    @GetMapping("/add")
-    public ModelAndView getInvestmentInterestForm() {
-        ModelAndView model = new ModelAndView("invinterest/add_invinterest");
-        model.addObject("invinterest", new InvestmentInterest());
-        return model;
+    @GetMapping("{id}")
+    public ResponseEntity<DtoInterest> getInterestById(@PathVariable long id) {
+        DtoInterest dtoInterest = investmentInterestService.getDtoInterestById(id);
+        if (dtoInterest != null) {
+            return new ResponseEntity<>(dtoInterest, HttpStatus.OK);
+        }
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
     }
 
-    @PostMapping("/add")
-    public ModelAndView addInvestmentInterest(@Valid InvestmentInterest investmentInterest,
-                                              BindingResult bindingResult,
-                                              Principal principal) {
-        if (bindingResult.hasErrors()) {
-            return new ModelAndView("invinterest/add_invinterest");
+    @PreAuthorize("isAuthenticated()")
+    @PutMapping("{id}")
+    public ResponseEntity<DtoInterest> updateInterest(
+            @PathVariable("id") long id,
+            @Valid @RequestBody DtoInterest dtoInterest) {
+        if (userService.getLoggedInUser().getId() != dtoInterest.getInvestor().getId()) {
+            return new ResponseEntity<>(HttpStatus.FORBIDDEN);
         }
-        User investor = userService.getUserByEmail(principal.getName());
-        investmentInterest.setInvestor(investor);
-        this.investmentInterestService.saveInvestmentInterest(investmentInterest);
-        return new ModelAndView(REDIRECT_TO_INVESTMENT_INTEREST);
+        DtoInterest currentInterest = investmentInterestService.updateDtoInterest(dtoInterest);
+        if (currentInterest != null) {
+            return new ResponseEntity<>(dtoInterest, HttpStatus.OK);
+        }
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
     }
 
-    @GetMapping("/{id}/delete")
-    public ModelAndView deleteInvestmentInterest(@PathVariable("id") Long id) {
-        this.investmentInterestService.deleteInvestmentInterest(id);
-        return new ModelAndView(REDIRECT_TO_INVESTMENT_INTEREST);
+    @PostMapping()
+    public ResponseEntity<DtoInterest> createInterest(@Valid @RequestBody DtoInterest dtoInterest) {
+        dtoInterest = investmentInterestService.createNewInterestFromDto(dtoInterest);

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "dtoInterest". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1226)